dspace-server-webapp/src/test/java/org/dspace/app/rest/RequestItemRepositoryIT.java

```diff
@@ -0,0 +1,368 @@
+     */
+    @Test
+    public void testFindOneNotAuthenticated()
+            throws Exception {
+        System.out.println("findOne");
+
+        context.turnOffAuthorisationSystem();
+
+        // Create necessary supporting objects.
+        Collection collection = CollectionBuilder.createCollection(context, parentCommunity)
+                .build();
+        Item item = ItemBuilder.createItem(context, collection)
+                .build();
+        InputStream is = new ByteArrayInputStream(new byte[0]);
+        Bitstream bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                .build();
+
+        // Create a request.
+        RequestItem request = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .build();
+
+        // Test:  can we find it?
+        final String uri = URI_ROOT + '/' + request.getToken();
+        getClient().perform(get(uri))
+                   .andExpect(status().isOk()) // Can we find it?
+                   .andExpect(content().contentType(contentType))
+                   .andExpect(jsonPath("$", Matchers.is(
+                       RequestCopyMatcher.matchRequestCopy(request))));
+
+        // Clean up.
+        bitstream.setDeleted(true);
+        context.restoreAuthSystemState();
+    }
+
+    /**
+     * Test of createAndReturn method, with an authenticated user.
+     *
+     * @throws java.sql.SQLException passed through.
+     * @throws org.dspace.authorize.AuthorizeException passed through.
+     * @throws java.io.IOException passed through.
+     */
+    @Test
+    public void testCreateAndReturnAuthenticated()
+            throws SQLException, AuthorizeException, IOException, Exception {
+        System.out.println("createAndReturn");
+
+        context.turnOffAuthorisationSystem();
+
+        // Create some necessary objects.
+        Collection col = CollectionBuilder.createCollection(context,
+                parentCommunity).build();
+        Item item = ItemBuilder.createItem(context, col).build();
+        InputStream is = new ByteArrayInputStream(new byte[0]);
+        Bitstream bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                .withName("/dev/null")
+                .withMimeType("text/plain")
+                .build();
+
+        // Fake up a request in REST form.
+        RequestItemRest rir = new RequestItemRest();
+        rir.setBitstreamId(bitstream.getID().toString());
+        rir.setItemId(item.getID().toString());
+        rir.setRequestEmail(RequestItemBuilder.REQ_EMAIL);
+        rir.setRequestMessage(RequestItemBuilder.REQ_MESSAGE);
+        rir.setRequestName(RequestItemBuilder.REQ_NAME);
+        rir.setAllfiles(false);
+
+        // Create it and see if it was created correctly.
+        ObjectMapper mapper = new ObjectMapper();
+        String authToken = getAuthToken(admin.getEmail(), password);
+        MvcResult mvcResult = getClient(authToken)
```

Comment:
This entire `getClient()` call can be simplified.  Returning the entire `mvcResult` uses a lot of extra time/memory...and it's not needed when all you want is the token created. 

Here's an example of how to do this much cleaner:
https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/test/java/org/dspace/app/rest/CollectionGroupRestControllerIT.java#L159-L165

The key here is you want to use an `andDo()` clause to parse the value out of the response **without** having to save the entire response to a variable.  You can find many more examples in the codebase by searching for "andDo("

Reply:
Isn't the MvcResult built anyway, and merely a reference returned?  I don't see how we save anything but a single store.

However, to make it more like the code elsewhere, done.

Reply:
@mwoodiupui : a while back we found out that Tests which accessed MvcResult directly ran _much slower_ than those that used `andDo()`.  So, somehow, the `andDo()` is somehow much more performant.  I admit, I don't know the *why*, but I do know that's why we adopted this "best practice" & avoid accessing MvcResult directly.